.github/workflows/single-binary.yml

```diff
@@ -42,6 +42,8 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
is this going to work in the [other 2 cases](https://github.com/flyteorg/flyte/pull/3733/files#diff-0a982aadf4b90a408619b7857a4369473eb3e8ba041ac21858ec49d5f2c3e0fcR13-R16)?

Reply:
nope